File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/domain/converter/TspToTbspConverterTest.java
package com.tarbadev.witchcraft.domain.converter;

import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Java6Assertions.assertThat;

public class TspToTbspConverterTest {
  private TspToTbspConverter subject;

  @Before
  public void setUp() {
    subject = new TspToTbspConverter();
  }

  @Test
  public void convert() {
    Double tsp = 12.0;
    Double tbsp = 4.0;

    assertThat(subject.convert(tsp)).isEqualTo(tbsp);

    tsp = 3.0;
    tbsp = 1.0;

    assertThat(subject.convert(tsp)).isEqualTo(tbsp);
  }
}

File: WitchcraftApp/src/test/kotlin/com/tarbadev/witchcraft/converter/TspToTbspConverterTest.kt
package com.tarbadev.witchcraft.converter

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class TspToTbspConverterTest {
    private lateinit var tspToTbspConverter: TspToTbspConverter

    @BeforeEach
    fun setUp() {
        tspToTbspConverter = TspToTbspConverter()
    }

    @Test
    fun convert() {
        var tsp = 12.0
        var tbsp = 4.0

        assertEquals(tbsp, tspToTbspConverter.convert(tsp))

        tsp = 3.0
        tbsp = 1.0

        assertEquals(tbsp, tspToTbspConverter.convert(tsp))
    }
}
